db/postgresql-instance.yml

```diff
@@ -0,0 +1,219 @@
+# db/rds-instance.yml
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >-
+  AWS CloudFormation RDS Template: Template to create a PostgreSQL RDS instance.
+Parameters:
+  DBInstance:
+    Default: "name-me"
+    Description: >-
+      The database instance name.
+      When replacement is required, must be different than existing instance name.
+    Type: String
+    MinLength: 0
+    MaxLength: 64
+    AllowedPattern: ^$|[a-zA-Z][a-zA-Z0-9\-]*
+    ConstraintDescription: >-
+      optional, must begin with a letter and contain only alphanumeric characters or dash.
+  CreateDBName:
+    Default: ""
+    Description: The database name for create only
+    Type: String
+    MinLength: 0
+    MaxLength: 64
+    AllowedPattern: ^$|[a-zA-Z][a-zA-Z0-9\-_]*
+    ConstraintDescription: >-
+      must begin with a letter and contain only alphanumeric characters, dash,
+      or underscore.
+  DBUser:
+    NoEcho: true
+    Description: The database admin account username
+    Type: String
+    MinLength: 1
+    MaxLength: 16
+    AllowedPattern: ^[a-zA-Z][a-zA-Z0-9_]*$
+    ConstraintDescription: must begin with a letter and contain only alphanumeric characters.
+  DBPassword:
+    NoEcho: true
+    Description: The database admin account password
+    Type: String
+    MinLength: 1
+    MaxLength: 41
+    AllowedPattern: ^[a-zA-Z0-9]+$
+    ConstraintDescription: must contain only alphanumeric characters.
+  DBAllocatedStorage:
+    Default: 20
+    Description: The size of the database (Gb)
+    Type: Number
+    MinValue: 20
+    MaxValue: 16384
+    ConstraintDescription: must be between 5 and 16384Gb.
+  DBInstanceClass:
+    Description: The database instance type
+    Type: String
+    Default: db.t2.small
+    ConstraintDescription: must select a valid database instance type.
```

Comment:
Removed the list of db types here, since they are always changing. It seems the user can just look up the list of types and plug one in here.